subprojects/core-api/src/main/java/org/gradle/api/artifacts/dsl/DependencyHandler.java

```diff
@@ -554,7 +555,23 @@ public interface DependencyHandler extends ExtensionAware {
      * @see TransformAction
      * @since 5.3
      */
-    <T extends TransformParameters> void registerTransform(Class<? extends TransformAction<T>> actionType, Action<? super TransformSpec<T>> registrationAction);
+    default <T extends TransformParameters> void registerTransform(Class<? extends TransformAction<T>> actionType, Action<? super TransformSpec<T>> registrationAction) {
```

Comment:
Do you know how this affects binary compatibility? 

I see the accepted public API changes has an entry for this change. Have we verified this is not a binary breaking change?

In any case, I think I would slightly prefer having both the old API (for backwards compatibility) and a new method that _always_ accepts a non-null name. 

I find it a bit odd that we have a method that _maybe_ accepts a name. I don't think there will be many cases where the user wants to optionally specify a name -- (programmatically that is, where they have some nullable name variable they are passing in)

I think this would be better as

```
    <T extends TransformParameters> void registerTransform(Class<? extends TransformAction<T>> actionType, Action<? super TransformSpec<T>> registrationAction)

    @Incubating
    <T extends TransformParameters> void registerTransform(String name, Class<? extends TransformAction<T>> actionType, Action<? super TransformSpec<T>> registrationAction);
```

Thoughts?